pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -22,18 +20,25 @@ pub struct EphemeralFile {
 
     _tenant_shard_id: TenantShardId,
     _timeline_id: TimelineId,
-    file: VirtualFile,
-    len: u64,
-    /// An ephemeral file is append-only.
-    /// We keep the last page, which can still be modified, in [`Self::mutable_tail`].
-    /// The other pages, which can no longer be modified, are accessed through the page cache.
+    /// We sandwich the buffered writer between two size-tracking writers.
+    /// This allows us to "elegantly" track in-memory bytes vs flushed bytes,
```

Comment:
"elegantly" :-D

Future: I think this could be made elegant though, if the sandwich was wrapped up in a separate type, and that separate type implemented a read() method that handled the read from the buffer.  Although we may find it makes more sense to just flush to disk if the data we want to read is still buffered, since that should be a pretty rare case (we might find there are frequent callers from wal ingest that should really be caching something at a higher level rather than reading-back very recent writes)